include/fmt/core.h

```diff
@@ -775,11 +780,12 @@ FMT_CONSTEXPR void basic_format_parse_context<Char>::check_dynamic_spec(
   }
 }
 
-template <typename Context> class basic_format_arg;
-template <typename Context> class basic_format_args;
-template <typename Context> class dynamic_format_arg_store;
+FMT_MODULE_EXPORT template <typename Context> class basic_format_arg;
+FMT_MODULE_EXPORT template <typename Context> class basic_format_args;
+FMT_MODULE_EXPORT template <typename Context> class dynamic_format_arg_store;
```

Comment:
These are just forward declarations. Do we still need to export them if the main templates are exported?

Reply:
Names must be declared `export` when **introduced** into a namespace [[module.interface]/6](http://eel.is/c++draft/module.interface#6). The compiler will then collect **all** semantic properties that pertain to that entity that is referenced by the name. This collection continues throughout the whole module purview up to the end of the TU or the start of the _private module fragment_, whichever comes first.

In this case, the (forward) _declarations_ introduce the names. So they need to be marked as exported at this very place in the source text.

This reminds me to remove the `export` declarations from the _definitions_ of the primary templates 🤦‍♂️